### ISSUE TITLE
Upgrade from 2014052104 to 2015071500 fails on PostgreSQL

### ISSUE THREAD

**Issue by mudrd8mz** [academic-moodle-cooperation/moodle-mod_checkmark#8]
The db/upgrade.php query near line 625 fails on PostgreSQL as column aliases can not be used in HAVING statements in this DB. This is not cross-db query:

```
SELECT ... COUNT( event.id ) AS present ... HAVING present = 0
```

The valid cross-db compatible form is

```
SELECT ... COUNT(event.id) AS present ... HAVING COUNT(event.id) = 0
```

**Comment by kaspot** [comment]
thanks for the info - is on our todo list.

**Comment by kaspot** [comment]
Should be fixed in the newest Moodle 2.8 version.
